File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio1/Java/main.java
package Generecidad.Ejercicio1.Java;

public class main {
    public static void main(String[] args) {
        // b) 

        // Instancia 1
        Caja<String> cajaString = new Caja<>(); 
        cajaString.guardar("Hola!");

        // Instancia 2
        Caja<Integer> cajaInteger = new Caja<>(); 
        cajaInteger.guardar(12345);

        // Instancia 3
        Caja<Double> cajaDouble = new Caja<>();
        cajaDouble.guardar(98.765);


        // c)
        System.out.println("--- Contenido de las Cajas ---");
        System.out.println("Caja de String: " + cajaString.obtener() + " (Tipo: " + cajaString.obtener().getClass().getName() + ")");
        System.out.println("Caja de Integer: " + cajaInteger.obtener() + " (Tipo: " + cajaInteger.obtener().getClass().getName() + ")");
        System.out.println("Caja de Double: " + cajaDouble.obtener() + " (Tipo: " + cajaDouble.obtener().getClass().getName() + ")");
        

        System.out.println("\n--- Usando el metodo toString() de la clase Caja ---");
        System.out.println(cajaString);
        System.out.println(cajaInteger);
        System.out.println(cajaDouble);
    }
}

File: repos_estudiantes_ALVAREZ_tercera_entrega/AJATA_CANAVIRI_NICOL_MAYRA/Generecidad/Ejercicio1/python/main.py
from Caja import Caja

if __name__ == "__main__":
    # b) Crea dos instancias de la caja y almacena 2 datos de diferente tipo

    # Instancia 1
    caja_entero = Caja[int]() 
    caja_entero.guardar(123)

    # Instancia 2
    caja_cadena = Caja[str]() 
    caja_cadena.guardar("Hola Mundo")

    # Instancia 3
    caja_flotante = Caja[float]()
    caja_flotante.guardar(3.14159)

    # c) Muestra el contenido de las cajas
    print("--- Contenido de las Cajas ---")
    print(f"Caja de Entero: {caja_entero.obtener()} (Tipo: {type(caja_entero.obtener())})")
    print(f"Caja de Cadena: {caja_cadena.obtener()} (Tipo: {type(caja_cadena.obtener())})")
    print(f"Caja de Float: {caja_flotante.obtener()} (Tipo: {type(caja_flotante.obtener())})")

    print("\n--- Usando el metodo __str__ de la clase Caja ---")
    print(caja_entero)
    print(caja_cadena)
    print(caja_flotante)